mmv1/third_party/terraform/services/compute/resource_compute_region_security_policy_test.go.tmpl

```diff
@@ -166,4 +166,699 @@ resource "google_compute_region_security_policy" "regionSecPolicy" {
+			name	= "tf-test%{random_suffix}"
+			type	= "CLOUD_ARMOR"
+			region  = "us-west2"
+
+			rules {
+				priority = "100"
+				action          = "throttle"
+				rate_limit_options {
+					conform_action = "allow"
+					exceed_action = "deny(429)"
+
+					rate_limit_threshold {
+						count = 10
+						interval_sec = 60
+					}
+
+					enforce_on_key_configs {
+						enforce_on_key_type = "USER_IP"
+					}
+
+					enforce_on_key_configs {
+						enforce_on_key_type = "TLS_JA3_FINGERPRINT"
+					}
+
+					enforce_on_key_configs {
+						enforce_on_key_type = "REGION_CODE"
+					}
+				}
+				match {
+					config {
+						src_ip_ranges = [
+							"*"
+						]
+					}
+					versioned_expr = "SRC_IPS_V1"
+				}
+			}
+
+			rules {
+				action   = "allow"
+				priority = "2147483647"
+				preview 	= false
+				match {
+					versioned_expr = "SRC_IPS_V1"
+					config {
+						src_ip_ranges = ["*"]
+					}
+				}
+				description = "default rule"
+			}
+		}
+	`, context)
+}
+
+func TestAccComputeRegionSecurityPolicy_regionSecurityPolicyWithRulesNetworkMatch(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckComputeRegionSecurityPolicyDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccComputeRegionSecurityPolicy_withNetworkMatch(context),
+			},
+			{
+				ResourceName:      "google_compute_region_security_policy.policy_network_match",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccComputeRegionSecurityPolicy_withNetworkMatch_update(context),
+			},
+			{
+				ResourceName:      "google_compute_region_security_policy.policy_network_match",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccComputeRegionSecurityPolicy_withNetworkMatch(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+	resource "google_compute_region_security_policy" "policy_ddos_enable" {
+		region      = "us-west1"
+		name        = "tf-test-ddos-enable%{random_suffix}"
+		description = "region security policy to pre-enable ddos protection in the region, otherwise network_match throws error ddos not enabled"
+		type        = "CLOUD_ARMOR_NETWORK"
+		ddos_protection_config {
+			ddos_protection = "ADVANCED_PREVIEW"
+		}
+	}
+
+	resource "google_compute_network_edge_security_service" "edge_security_service" {
+		region          = "us-west1"
+		name            = "tf-test-edge-service%{random_suffix}"
+		description = "edge security service is required to be pre-created when creasting a policy with network_match rules"
+		security_policy = google_compute_region_security_policy.policy_ddos_enable.self_link
+	}
```

Comment:
Are either of these resources "global" in a way where they could potentially impact other tests running in the same project? If so it would be good to create a new project in this test so that they're isolated.